pkgs/tools/archivers/p7zip/default.nix

```diff
@@ -22,40 +22,33 @@ stdenv.mkDerivation rec {
     '';
   };
 
-  # Default makefile is full of impurities on Darwin. The patch doesn't hurt Linux so I'm leaving it unconditional
-  postPatch = ''
-    sed -i '/CC=\/usr/d' makefile.macosx_llvm_64bits
-    # Avoid writing timestamps into compressed manpages
-    # to maintain determinism.
-    substituteInPlace install.sh --replace 'gzip' 'gzip -n'
-    chmod +x install.sh
-
-    # I think this is a typo and should be CXX? Either way let's kill it
-    sed -i '/XX=\/usr/d' makefile.macosx_llvm_64bits
-  '' + lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
-    substituteInPlace makefile.machine \
-      --replace 'CC=gcc'  'CC=${stdenv.cc.targetPrefix}gcc' \
-      --replace 'CXX=g++' 'CXX=${stdenv.cc.targetPrefix}g++'
-  '';
-
-  makeFlags = [ "DEST_HOME=${placeholder "out"}" ];
-
   preConfigure = ''
     buildFlags=all3
   '' + lib.optionalString stdenv.isDarwin ''
     cp makefile.macosx_llvm_64bits makefile.machine
+  '' + lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    substituteInPlace makefile.machine \
+      --replace 'CC=gcc'  'CC=${stdenv.cc.targetPrefix}gcc' \
+      --replace 'CXX=g++' 'CXX=${stdenv.cc.targetPrefix}g++'
   '';
 
   enableParallelBuilding = true;
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=c++11-narrowing";
 
-  setupHook = ./setup-hook.sh;
+  makeFlags = [
+    "DEST_BIN=${placeholder "out"}/bin"
+    "DEST_SHARE=${placeholder "lib"}/lib/p7zip"
+    "DEST_MAN=${placeholder "man"}/share/man"
+    "DEST_SHARE_DOC=${placeholder "doc"}/share/doc/p7zip"
+  ];
 
-  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=c++11-narrowing";
+  outputs = [ "out" "lib" "doc" "man" ];
```

Comment:
Outputs typically go between `version` and `src` https://github.com/jtojnar/nixpkgs-hammering/blob/243b81c687aac33d6716957c0cd2235c81631044/explanations/attribute-ordering.md

Reply:
> This is nothing but a personal preference but [growing stronger](https://discourse.nixos.org/t/document-attribute-ordering-in-package-expressions/4887).

With the link only having yourself mention outputs.

Respectfully I'd like to keep it here near where install related stuff tends to go